Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -346,13 +346,39 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
                                                                     operationDispatcher: operationDispatcher,
                                                                     api: backend.offlineEntitlements,
                                                                     systemInfo: systemInfo)
-        let customerInfoManager = CustomerInfoManager(offlineEntitlementsManager: offlineEntitlementsManager,
+
+        var diagnosticsFileHandler: DiagnosticsFileHandlerType?
+        var diagnosticsTracker: DiagnosticsTrackerType?
+        if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
+            if diagnosticsEnabled {
+                diagnosticsFileHandler = DiagnosticsFileHandler()
+                if let diagnosticsFileHandler = diagnosticsFileHandler {
+                    diagnosticsTracker = DiagnosticsTracker(diagnosticsFileHandler: diagnosticsFileHandler)
+                } else {
+                    Logger.error(Strings.diagnostics.could_not_create_diagnostics_tracker)
+                }
+            }
+        }
```

Comment:
Consider initializing these variables using closures, which allows us to define them as let constants.
 
```suggestion
        let diagnosticsFileHandler: DiagnosticsFileHandlerType? = {
            guard diagnosticsEnabled, #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) else { return nil }
            return DiagnosticsFileHandler()
        }()

        let diagnosticsTracker: DiagnosticsTrackerType? = {
            if let handler = diagnosticsFileHandler, #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
                return DiagnosticsTracker(diagnosticsFileHandler: handler)
            } else {
                if diagnosticsEnabled {
                    Logger.error(Strings.diagnostics.could_not_create_diagnostics_tracker)
                }
            }
            return nil
        }()
```

Reply:
That's very nice. I knew there had to be a cleaner way haha this is it!